apps/web/components/posts/table.tsx

```diff
@@ -1,12 +1,24 @@
 import { useRouter } from "next/router";
-import { Post } from "../../interface/post";
+import { Post, DeletePostResult } from "../../interface/post";
+import { gql, useMutation, DocumentNode } from "@apollo/client";
 
 interface TableProps {
   posts: Post[];
+  GET_POSTS: DocumentNode
 }
 
-const Table = ({ posts }: TableProps) => {
+const DEL_POST = gql`
+  mutation Delete($id: ID!) {
+  	deletePost(id: $id)
+  }
+`;
+
+const Table = ({ posts, GET_POSTS }: TableProps) => {
   const { push } = useRouter();
+  const [deletePost] = useMutation<DeletePostResult>(DEL_POST, {refetchQueries: [GET_POSTS]});
```

Comment:
[nit] If we also create an interface for the mutation variables and use them like this
```suggestion
  const [deletePost] = useMutation<DeletePostResult, DeletePostVariables>(...);
```
we get type hints when we call `deletePost({ variables: { id: id }})`. Otherwise, we have to remember what the query variable names should be.